src/components/Modals/WorldPermissionsModal/WorldPermissionsModal.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { Button, ButtonProps, CheckboxProps, InputOnChangeData, ModalActions, ModalContent, ModalNavigation, Tabs } from 'decentraland-ui'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import LoadingText from 'decentraland-ui/dist/components/Loader/LoadingText'
+import { isValid } from 'lib/address'
+import { Props } from './WorldPermissionsModal.types'
+import { AllowListPermissionSetting, WorldPermissionNames, WorldPermissionType } from 'lib/api/worlds'
+import WorldPermissionsAccess from './ModelTabs/WorldPermissionsAccess/WorldPermissionsAccess'
+
+import './WorldPermissionsModal.css'
+import WorldPermissionsCollaborators from './ModelTabs/WorldPermissionsCollaborators/WorldPermissionsCollaborators'
+
+enum TabsSection {
+  Access = 'Access',
+  Collaborators = 'Collaborators'
+}
+
+const WorldPermissionsModal = (props: Props) => {
+  const {
+    name,
+    metadata,
+    worldPermissions,
+    profiles,
+    isLoading,
+    isLoadingNewUser,
+    onPutWorldPermissionsRequest,
+    onPostWorldPermissionsRequest,
+    onDeleteWorldPermissionsRequest,
+    onGetProfilesRequest,
+    onClose
+  } = props
+  const [newAddress, setNewAddress] = useState('')
+  const [showAddUserForm, setShowAddUserForm] = useState(false)
+  const [errorInvalidAddress, setErrorInvalidAddress] = useState(false)
+  const [isAccessUnrestricted, setAccessUnrestricted] = useState(worldPermissions?.access.type === WorldPermissionType.Unrestricted)
+  const [collaboratorUserList, setCollaboratorUserList] = useState<string[]>([])
+  const [tabSelected, setTabSelected] = useState<TabsSection>(TabsSection.Access)
+
+  const loading = isLoading || !metadata.worldName || !worldPermissions
+
+  useEffect(() => {
+    setAccessUnrestricted(worldPermissions?.access.type === WorldPermissionType.Unrestricted)
+  }, [worldPermissions?.access.type])
+
+  useEffect(() => {
+    if (
+      tabSelected === TabsSection.Access &&
+      worldPermissions?.access.type === WorldPermissionType.AllowList &&
+      worldPermissions?.access.wallets.length > 0
+    ) {
+      const allowedWallets = new Set(worldPermissions.access.wallets)
+      allowedWallets.has(newAddress.toLowerCase()) && setNewAddress('')
```

Comment:
We usually use `if` conditions when there's the need to execute code, would you mind changing it?
```suggestion
      if (allowedWallets.has(newAddress.toLowerCase())) {
        setNewAddress('')
      }
```